redfish-core/lib/systems.hpp

```diff
@@ -51,6 +55,56 @@
 namespace redfish
 {
 
+/* @brief validate systemName and get system index from systemName
+ *
+ * @param[in] systemName     requested system name
+ *
+ * @return int    -1 on failure, else validatet index
+ * */
+inline int getSystemIndex(const std::string_view systemName)
+{
+#ifndef BMCWEB_ENABLE_MULTI_COMPUTERSYSTEM
```

Comment:
personal preference, but IMO it would be prettier to reduce nesting and make 2 funtions here.
getSystemIndexSingleHost and getSystemIndexMultiHost, and use the config option to select which.

Reply:
As this function is called several times in other functions, wouldn't i have to write a check for the option in every calling function? As of now I only have to write the check once in getSystemIndex()

Reply:
As discussed, i changed the implementation of getSystemIndex, in order to push for removing the experimental meson option for multi-host.